src/BugsnagUnity.m

```diff
@@ -80,10 +80,22 @@ void bugsnag_setEventMetadata(const void *event, const char *tab, const char *me
     return ((__bridge BugsnagSession *)session).user;
 }
 
+void bugsnag_setUserFromSession(const void *session, char *userId, char *userName, char *userEmail){
+    [((__bridge BugsnagSession *)session) setUser:userId == NULL ? nil : [NSString stringWithUTF8String:userId]
+            withEmail:userEmail == NULL ? nil : [NSString stringWithUTF8String:userEmail]
+             andName:userName == NULL ? nil : [NSString stringWithUTF8String:userName]];
+}
+
 BugsnagUser * bugsnag_getUserFromEvent(const void *event){
     return ((__bridge BugsnagEvent *)event).user;
 }
 
+void bugsnag_setUserFromEvent(const void *event, char *userId, char *userName, char *userEmail){
+    [((__bridge BugsnagEvent *)event) setUser:userId == NULL ? nil : [NSString stringWithUTF8String:userId]
+            withEmail:userEmail == NULL ? nil : [NSString stringWithUTF8String:userEmail]
+             andName:userName == NULL ? nil : [NSString stringWithUTF8String:userName]];
+}
+
```

Comment:
I'm going to guess that these might need some tests? Its weird that they could be completely absent without error.

Reply:
_Resolving via discussion of the tests in subsequent changeset_